include/seastar/rpc/rpc_types.hh

```diff
@@ -342,8 +342,8 @@ public:
     future<> operator()(const Out&... args) {
         return _impl->operator()(args...);
     }
-    future<> close() {
-        return _impl->close();
+    future<> close() noexcept {
+        return futurize_invoke([this] { return _impl->close(); });
```

Comment:
IMO, better to make impl::close() noexcept and audit implementations.

Reply:
just thay it would be an an api change, though minor so maybe doesn't require an api level.